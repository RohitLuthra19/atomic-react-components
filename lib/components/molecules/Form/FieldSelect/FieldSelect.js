// @flow
import React from 'react';
import type { Node } from 'react';
import { Field, ErrorMessage } from 'formik';
import styled from 'styled-components';
import Label from '../../../atoms/Label';
import Select from '../../../atoms/Select';
import FieldError from '../../../atoms/FieldError';
import styles from './FieldSelect.style';


type Props = {
  id: string,
  className: string,
  name: string,
  label: Node,
  labelProps ?: Object,
  fieldProps ?: Object,
};

const FieldSelect = ({
  id,
  className,
  name,
  label,
  labelProps,
  fieldProps,
  ...others
}: Props): Node => (
  <div className={className}>
    <Label htmlFor={id} {...labelProps}>{label}</Label>
    <Field
      name={name}
      component="select"
      render={({ field }) => <Select {...others} {...field} />}
      {...fieldProps}
    />
    <ErrorMessage name={name} component={FieldError} />
  </div>
);

FieldSelect.defaultProps = {
  labelProps: {},
  fieldProps: {},
};

export default styled(FieldSelect)`${styles}`;
export { FieldSelect as FieldSelectVanilla };
