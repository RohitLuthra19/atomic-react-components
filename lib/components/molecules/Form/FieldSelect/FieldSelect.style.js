import { css } from 'styled-components';

const styles = css`
  margin-bottom: 1rem;

  label {
    display: block;
    margin-bottom: 0.25rem;
  }
`;

export default styles;
