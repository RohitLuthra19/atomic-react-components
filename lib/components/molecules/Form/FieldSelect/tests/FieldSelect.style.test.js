import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import FieldSelect from '../FieldSelect';

describe('FieldSelect', () => {
  test('should render correctly', () => {
    expect(shallow(<FieldSelect />)).toMatchSnapshot();
  });
});
