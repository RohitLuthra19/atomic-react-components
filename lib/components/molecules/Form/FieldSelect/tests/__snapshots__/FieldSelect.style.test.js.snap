// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldSelect should render correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "FieldSelect-sc-1fkkiga-0",
        "isStatic": false,
        "rules": Array [
          "margin-bottom:1rem;label{display:block;margin-bottom:0.25rem;}",
          ";",
        ],
      },
      "displayName": "FieldSelect",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "FieldSelect-sc-1fkkiga-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
